File: DeveloperJobApplicationTwo/Model/DeveloperJob.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace DeveloperJobApplicationTwo.Model
{
    public class DeveloperJob
    {
        public int Id { get; set; }
        public string JobTitle { get; set; }
        public string CompanyName { get; set; }
        public virtual CompanyAddress CompanyAddress { get; set; }
        public string SalaryRange { get; set; }
        public bool? RemoteOptions { get; set; }
        public bool? EquitySharing { get; set; }
        public JobType JobType { get; set; }
        public int JobTypeId { get; set; }
        public string ExperienceLevel { get; set; }
        public Industry Industry { get; set; }
        public int IndustryId { get; set; }
        public int CompanySize { get; set; }
        public CompanyType CompanyType { get; set; }
        public int CompanyTypeId { get; set; }
        public string JobDescription { get; set; }

        
        public bool IsActive { get; set; }
        public DateTime? CreatedDate { get; set; }
        public DateTime? DeactivatedDate { get; set; }
    }
}

File: DeveloperJobApplicationTwo/ClientApp/app/model/developerjob.ts


export class CompanyAddress {

    constructor(

        public id: number,
        public address1: string,
        public address2: string,
        public city: string,
        public state: string,
        public zipCode: string,
        public countryName:string

    ) { }
}

export class DeveloperJob {

    constructor(

        public id: number,
        public jobTitle: string,
        public companyName: string,
        public companyAddress: CompanyAddress,
        public salaryRange: string,
        public remoteOptions: boolean,
        public equitySharing: boolean,
        public jobTypeId: number,
        public experienceLevel: string,
        public industryId: number,
        public companySize: number,
        public companyTypeId: number,
        public jobDescription: string


    ) { }
}

export class DeveloperJobView {

    constructor(

        public developerJobId: number,
        public jobTitle: string,
        public companyName: string,
        public companyAddress: CompanyAddress,
        public salaryRange: string,
        public jobTypeName: string,
        public experienceLevel: string,
        public industryName: string,
        public companySize: number,
        public companyTypeName: string,

        public createdDate: Date,
        public isNewJob: boolean
        
    ) { }
}


